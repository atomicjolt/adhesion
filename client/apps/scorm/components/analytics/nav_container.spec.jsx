import React from 'react';
import { shallow } from 'enzyme';
import NavContainer from './nav_container';

describe('Scorm Analytics NavContainer', () => {

  let result;
  let switched = false;
  const props = {
    navButtons: [
      {
        name: 'test application',
        stat: '12',
      },
    ],
    switchChart: () => { switched = true; },
  };

  beforeEach(() => {
    result = shallow(<NavContainer {...props} />);
  });

  it('match the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
