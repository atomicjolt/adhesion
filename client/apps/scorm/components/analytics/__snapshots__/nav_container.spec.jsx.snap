// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorm Analytics NavContainer match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavButtons
    navButtons={
        Array [
            Object {
              "name": "test application",
              "stat": "12",
            },
          ]
    }
    switchChart={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavButton
          activeBtn="complete"
          label="test application"
          setActive={[Function]}
          stat="12"
          type={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "nav_button_test application",
        "nodeType": "function",
        "props": Object {
          "activeBtn": "complete",
          "label": "test application",
          "setActive": [Function],
          "stat": "12",
          "type": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavButton
            activeBtn="complete"
            label="test application"
            setActive={[Function]}
            stat="12"
            type={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "nav_button_test application",
          "nodeType": "function",
          "props": Object {
            "activeBtn": "complete",
            "label": "test application",
            "setActive": [Function],
            "stat": "12",
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
